src/Queen.cpp

```diff
@@ -0,0 +1,19 @@
+#include"Queen.h"
+
+Queen::Queen(ChessPoint _pos, bool _is_white) :Piece('q', _pos, _is_white){}
+
+
+std::vector<ChessPoint> Queen::allLegalMoves(const ChessBoard& board) const
+{
+    std::vector<ChessPoint> moves;
+
+    // Compute moves from the Rook
+    std::vector<ChessPoint> rook_moves = Rook::rookMoves(this, board);
```

Comment:
Intermediate classes must be created for walking diagonally and walking straight, thus giving a finer inheritance for different tools. instead of calling the rook and bishop moves

Reply:
During the class discussion, different approaches to handle movement variations in chess pieces were discussed. One option suggested using intermediate classes for diagonal and straight movement, but it was acknowledged that this approach had drawbacks regarding the Liskov Substitution Principle. In my implementation, I carefully evaluated the alternatives and decided that my chosen approach was the most appropriate. The lecturer confirmed that incorporating the alternative method was not required for scoring purposes.